### ISSUE TITLE
Zoomit.ir cosmetic filtering

### ISSUE THREAD

**Issue by nimasaj** [uBlockOrigin/uAssets#3959]
### URL(s) where the issue occurs

`https://www.zoomit.ir/`

### Describe the issue

Expandable ads place holder on header section of site, as it could be seen through following screenshot. It is loaded in `<div class=declarations>` in source code of the page. In DevTools, it's shown in `https://cdn.pbrd.co/images/HKGi4kB.png`.

### Screenshot(s)
Ads place holder screenshot:
`https://cdn.pbrd.co/images/HKGdXqp.png`

### Versions

- Browser/version: 70.0.3538.67 (Official Build) (64-bit)
- uBlock Origin version: 1.17.0

### Settings

- Default + `http://mynext.pro/uBOPa.txt`

### Notes

I tried with following rule, but it is not helpful, and ads place holder will be left blank.
`zoomit.ir##div[class^="declaration-wrapper"]`

**Comment by nimasaj** [comment]
By not using filter list, the issue is not gone. It just shows ads and after some reloads the placeholder is gone by itself. Try to load the page in incognito mode and check again with and without uBO.

An image with disabled uBO is available at:
`https://cdn.pbrd.co/images/HKJkQ4x.png`

**Comment by kulfoon** [comment]
Yes, I notice now, I thought it's an image/logo and not an ad.

**Comment by kulfoon** [comment]
Enable one of the regional list and try this:
```
zoomit.ir##body:style(padding-top: 0% !important;)
zoomit.ir##.is-sticky.scrollSticky.navbar-margin.navbar-default.navbar:style(position: static !important;)
```

**Comment by ZaphodBeebblebrox** [comment]
With Adblock Iran enabled, does 
```
zoomit.ir##body:style(padding-top: 0px !important;)
zoomit.ir##.navbar:style(margin-top: 0px !important;)
```
work for you?

**Comment by krystian3w** [comment]
As for me, only the script rule would do it better, but does anyone have so much time to analyze ...

**Comment by ZaphodBeebblebrox** [comment]
I could do something like `zoomit.ir##+js(abort-on-property-read.js, setHeightOfAd)` but I am afraid that I might break somethign important, and its hard to tell for certain for a website in a language I do not understand.

**Comment by krystian3w** [comment]
> `zoomit.ir##.navbar:style(margin-top: 0px !important;)`

seem does not work, no move up `.navbar`. For me.

**Comment by ZaphodBeebblebrox** [comment]
And the option in my other comment, does it work?

**Comment by krystian3w** [comment]
> `zoomit.ir##body:style(padding-top: 0px !important;)`

almost covers  the advert only so that you need to navigate this navigation nicely / skillfully.

**Comment by ZaphodBeebblebrox** [comment]
I was referring to `zoomit.ir##+js(abort-on-property-read.js, setHeightOfAd)`

**Comment by okiehsch** [comment]
`zoomit.ir##body:style(padding-top: 10px !important;)` works for me.

**Comment by krystian3w** [comment]
`zoomit.ir##body:style(padding-top: 0px !important;)`
`zoomit.ir##.hotMenu:style(top: 0 !important;)`

I do not know Iranian enough to test what broke down on the script rule.

```
||zoomit.ir/Themes/Zoomit.ir/Css/img/declarationPics/image-*$image,first-party
||cdn01.zoomit.ir/banners/videos/*$media,first-party
```

this is go to Iran List?

**Comment by ZaphodBeebblebrox** [comment]
Does that rule not leave you with a ugly bar?
![image](https://user-images.githubusercontent.com/22801430/47680920-8d7e9900-db9e-11e8-8b4b-31d0bb5877ce.png)

**Comment by krystian3w** [comment]
Nope:

![obraz](https://user-images.githubusercontent.com/35370833/47681031-b451c500-dbc8-11e8-89c2-eb2ae0a606b5.png)


Maybe bug in my Opera 56...

**Comment by ZaphodBeebblebrox** [comment]
Eh, it appears to work well enough. We can probably just add it.

**Comment by okiehsch** [comment]
@farrokhi do you want to add the filter
`zoomit.ir##body:style(padding-top: 10px !important;)`
 to your list ?

**Comment by kulfoon** [comment]
Forget these:
`||zoomit.ir/Themes/Zoomit.ir/Css/img/declarationPics/image-*$image,first-party`
`||cdn01.zoomit.ir/banners/videos/*$media,first-party`

I just missed that they are already blocked in both regional IRA lists,

**Comment by nimasaj** [comment]
Thanks for all replies.
None of the following ones worked for me. It blocks as long as you're not scrolling. after scrolling it will be like `https://cdn.pbrd.co/images/HKSRutP.png`.


`zoomit.ir##body:style(padding-top: 10px !important;)`

```
zoomit.ir##body:style(padding-top: 0px !important;)
zoomit.ir##.hotMenu:style(top: 0 !important;)
```

```
||zoomit.ir/Themes/Zoomit.ir/Css/img/declarationPics/image-*$image,first-party
||cdn01.zoomit.ir/banners/videos/*$media,first-party
```

```
zoomit.ir##body:style(padding-top: 0px !important;)
zoomit.ir##.navbar:style(margin-top: 0px !important;)
```

The only filter that worked for me was `zoomit.ir##+js(abort-on-property-read.js, setHeightOfAd)` 
Thanks to @ZaphodBeebblebrox

**Comment by farrokhi** [comment]
Just merged into Adblock-Iran list. Thanks to everyone for the suggestions and report.

**Comment by nimasaj** [comment]
@ZaphodBeebblebrox Could you please mention where did you find parameter `setHeightOfAd` in filter `zoomit.ir##+js(abort-on-property-read.js, setHeightOfAd)`?
I cannot find it neither in page source, nor in scripts loaded with the page.

**Comment by mapx-** [comment]
in an external script 
`https://www.zoomit.ir/abundle/scripts/zoomitqos....`

**Comment by kulfoon** [comment]
The ZaphodBeebblebrox ' solution has an side-effect - it changes default behaviour of the header, the header gets unsticky and is not scrolling when scrolling the webpage. 

@nimasaj is that behaviour desirable for you? Or you like more to keep the header's default behaviour?

Anyway, I spent some more time with the issue and found the both alternative (to script injection) solutions,  could anyone test these:

Solution 1: is an equivalent of the ZaphodBeebblebrox ' solution:
```
! unsticky solution
zoomit.ir##.declarations
zoomit.ir##body:style(padding-top: 0px !important;)
zoomit.ir##.hotMenu:style(top: 0px !important;)
zoomit.ir##.navbar.navbar-default:style(top: 0px !important; position: static !important;)
```

Solution 2: restores the original behavior of the header
```
! sticky solution
zoomit.ir##.declarations
zoomit.ir##body:style(padding-top: 0px !important;)
zoomit.ir##.hotMenu:style(top: 0px !important;)
zoomit.ir##.navbar.navbar-default:style(top: 0px !important;)
```

**Comment by krystian3w** [comment]
Colose ad become unsticky navigation.

**Comment by kulfoon** [comment]
?

**Comment by nimasaj** [comment]
@kulfoon , You're right. sticky header is the original behavior and better to keep it like that. Thanks for finding alternative solutions. I will update it in uBOPa list.
